src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java

```diff
@@ -128,13 +128,17 @@ public void start(final Map<String, String> parsedConfig)
         SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC_DEFAULT));
     }
 
-    connectorName = Utils.validateConfig(config);
+    Utils.validateConfig(config);
+
+    // modify invalid connector name
+    connectorName = Utils.convertAppName(config);
 
     // create a persisted connection, and validate snowflake connection
     // config as a side effect
     conn = SnowflakeConnectionServiceFactory
       .builder()
       .setProperties(parsedConfig)
+      .setConnectorName(connectorName)
```

Comment:
if replace _parsedConfig_ by _config_ here, may not need to call setConenctorName() method.

Reply:
ack